src/test/lib/DataTest.cpp

```diff
@@ -512,6 +542,42 @@ QuicTestConnectAndPing(
             TEST_FAILURE("Wait for server to complete timed out after %u ms.", TimeoutMs);
             return;
         }
+
+        if (ClientSecrets) {
+            for (auto i = 0u; i < ConnectionCount; i++) {
+                auto ServerSecret = &ServerSecrets[i];
+                bool Match = false;
+                for (auto j = 0u; j < ConnectionCount; j++) {
+                    auto ClientSecret = &ClientSecrets[j];
+                    if (!memcmp(
+                            ServerSecret->ClientRandom,
+                            ClientSecret->ClientRandom,
+                            sizeof(ClientSecret->ClientRandom))) {
+                        if (Match) {
+                            TEST_FAILURE("Multiple clients with the same ClientRandom?!");
+                            return;
+                        }
```

Comment:
This is theoretically possible, right? Just HIGHLY unlikely. Should we fail the tests if _somehow_ it happens?